--- ./package/libs/openssl/Makefile	2015-01-19 23:17:03.000000000 +0800
+++ ./package/libs/openssl/Makefile	2015-01-19 23:17:41.535521063 +0800
@@ -24,7 +24,7 @@ PKG_LICENSE:=SSLEAY OPENSSL
 PKG_LICENSE_FILES:=LICENSE
 PKG_BUILD_DEPENDS:=ocf-crypto-headers
 PKG_CONFIG_DEPENDS:=CONFIG_OPENSSL_ENGINE_CRYPTO CONFIG_OPENSSL_ENGINE_DIGEST \
-	CONFIG_OPENSSL_WITH_EC CONFIG_OPENSSL_WITH_EC2M
+	CONFIG_OPENSSL_WITH_EC CONFIG_OPENSSL_WITH_EC2M CONFIG_OPENSSL_THREADS
 
 include $(INCLUDE_DIR)/package.mk
 
@@ -80,7 +80,13 @@ endef
 
 OPENSSL_NO_CIPHERS:= no-idea no-md2 no-mdc2 no-rc5 no-sha0 no-smime \
 	no-aes192 no-camellia no-ans1 no-krb5
-OPENSSL_OPTIONS:= shared no-err no-hw zlib-dynamic no-sse2
+OPENSSL_OPTIONS:= shared no-err no-hw zlib-dynamic no-sse2 no-ssl2 no-ssl3 no-rc5 no-mdc2 no-idea
+
+ifdef CONFIG_OPENSSL_THREADS
+  OPENSSL_OPTIONS += threads
+else
+  OPENSSL_OPTIONS += no-threads
+endif
 
 ifdef CONFIG_OPENSSL_ENGINE_CRYPTO
   OPENSSL_OPTIONS += -DHAVE_CRYPTODEV
@@ -107,8 +113,12 @@ else
   ifeq ($(CONFIG_mips)$(CONFIG_mipsel),y)
     OPENSSL_TARGET:=linux-mips-openwrt
   else
-    OPENSSL_TARGET:=linux-generic-openwrt
-    OPENSSL_OPTIONS+=no-perlasm
+    ifeq ($(CONFIG_powerpc),y)
+      OPENSSL_TARGET:=linux-ppc-openwrt
+    else
+      OPENSSL_TARGET:=linux-generic-openwrt
+      OPENSSL_OPTIONS+=no-perlasm
+    endif
   endif
 endif
 
@@ -144,7 +154,7 @@ TARGET_CFLAGS += $(FPIC)
 define Build/Compile
 	+$(MAKE) $(PKG_JOBS) -C $(PKG_BUILD_DIR) \
 		CC="$(TARGET_CC)" \
-		ASFLAGS="$(TARGET_ASFLAGS) -c" \
+		ASFLAGS="$(TARGET_ASFLAGS) -I$(PKG_BUILD_DIR)/crypto -c" \
 		AR="$(TARGET_CROSS)ar r" \
 		RANLIB="$(TARGET_CROSS)ranlib" \
 		OPENWRT_OPTIMIZATION_FLAGS="$(TARGET_CFLAGS)" \
@@ -152,7 +162,7 @@ define Build/Compile
 		all
 	+$(MAKE) $(PKG_JOBS) -C $(PKG_BUILD_DIR) \
 		CC="$(TARGET_CC)" \
-		ASFLAGS="$(TARGET_ASFLAGS) -c" \
+		ASFLAGS="$(TARGET_ASFLAGS) -I$(PKG_BUILD_DIR)/crypto -c" \
 		AR="$(TARGET_CROSS)ar r" \
 		RANLIB="$(TARGET_CROSS)ranlib" \
 		OPENWRT_OPTIMIZATION_FLAGS="$(TARGET_CFLAGS)" \
